Meta/build-image-qemu.sh

```diff
@@ -31,7 +31,17 @@ if [ "$(id -u)" != 0 ]; then
     if [ -x "$FUSE2FS_PATH" ] && $FUSE2FS_PATH --help 2>&1 |grep fakeroot > /dev/null; then
         USE_FUSE2FS=1
     else
-        ${SUDO} -E -- "$0" "$@" || die "this script needs to run as root"
+        ${SUDO} -E -- sh -c "$0"' '"$*"'|| exit 42' || \
```

Comment:
The quotation marks make this a bit hard to read. Don't we have some other way to construct the same string?

Reply:
That said, I'm pretty sure that what we are doing here messes up the string splitting anyways, so we might as well do the following:

```suggestion
        ${SUDO} -E -- sh -c "$0 $* || exit 42" || \
```

Reply:
is $* even needed here? looks like the script doesn't take any arguments.
$0 not having quotes might eventually error out if someone tries to execute the script manually with its full path... which is rare, but nevertheless a case to be taken into account...

Reply:
I don't think there is any way to preserve spaces in arguments when using `sh -c`, unless one wants to go through the effort of manually wrapping each part in `'` or `"`.

The reason why `$*` is there at all is because this is a pretty generic code snippet for "how do I rerun a script with root permissions". But in the end, I don't think it hurts any, we'd have to be concerned with spaces either way because of `$0`.

Reply:
quoted $0 in the latest changes, but not $* as im not sure that's really required